### ISSUE TITLE
422 HTTP code on provider selection

### ISSUE THREAD

**Issue by nsklikas** [canonical/identity-platform-login-ui#41]
When the user gets redirected to the login page, they are presented with a set of providers to choose from. When they click on a provider, an XHR request is done to the `/self-service/login` endpoint. The backend responds with a 422 code and a URL to redirect to (the authn endpoint).

This happens because the application type we send is `application/json`, if the application type is changed to `application/x-www-form-urlencoded`, the status code should change to `303`. It is not enough to add that content type to the frontend headers, as the ory go client does not allow us to set the content type when making the request (at least I couldn't figure out how to do it).

We could handle this in the backend (ie convert the 422 to a 303 as long as the response body is valid).

**Comment by amancooks08** [comment]
Absolutely, could you assign me this issue, I'm new to this ecosystem, would love to contribute!
